### PR TITLE
Implement Pascal Case for Erc20Permit and Erc721Permit

### LINKED ISSUE #39: Three Letter Linter
Make a custom lint rule enforcing this. Check previous conversation 

> Our coding style guide should have this written as `Erc20Permit`

> It is an implicit rule according to the contributing.md in the Ubiquity Dollar repo

_Originally posted by @0x4007 in https://github.com/ubiquity/pay.ubq.fi/pull/224#discussion_r1569165738_

### PR DESCRIPTION
**Pull request by jordan-ae**
… naming conventions

Resolves #39

### CODE REVIEW

**Comment by 0x4007** [comment]
Thank you but this should be generalized for every identifier not just those two.

File: .eslintrc [hunk 1]
```diff
@@ -49,7 +49,15 @@
       { "selector": "variable", "modifiers": ["destructured"], "format": null },
       { "selector": "variable", "types": ["boolean"], "format": ["PascalCase"], "prefix": ["is", "should", "has", "can", "did", "will", "does"] },
       { "selector": "variableLike", "format": ["camelCase"] },
-      { "selector": ["function", "variable"], "format": ["camelCase"] }
+      { "selector": ["function", "variable"], "format": ["camelCase"] },
+      {
+        "selector": "variable",
+        "format": ["PascalCase"],
+        "filter": {
+          "regex": "^(Erc20Permit|Erc721Permit)$",
```
Comment (0x4007) [inline .eslintrc]: Thank you for your pull but this should be generalized for every identifier not just those two.

**Review by 0x4007 — CHANGES_REQUESTED** [review]
You also should not change any other files.

**Comment by jordan-ae** [comment]
> You also should not change any other files. 

@0x4007 The other modifications were made my running the linters. If you check the changes you'll see the linters just added spaces where it seemed necessary and all that. I can revert the changes if they will cause any issues though.

File: .eslintrc (hunk 1 above)
Comment (jordan-ae) [reply .eslintrc]: If I understand what you mean by identifier: variables, types, functions, and labels should all follow pascal case?

**Comment by gentlementlegen** [comment]
There actually already is a rule for `PascalCase` interfaces, no need to create a new one. The thing is that rule enforces `Pascal Case` but not strictly meaning you can chain uppercase letters.

If we want to prevent this we can use `StrictPascalCase`. Maybe it would be a good idea to use it everywhere @0x4007 ?
See also: https://typescript-eslint.io/rules/naming-convention/#format-options

**Comment by 0x4007** [comment]
That rule seems to only check for the first character being capitalized according to the docs. I think a regex that looks for three or more consecutive capitalized letters should be flagged as an error

**Comment by gentlementlegen** [comment]
`StrictPascalCase` will enforce no more that one capital letter inside the name, regardless of the position. Example:
image
Quoting the docs: "consecutive capitals are not allowed (i.e. myId is valid, but myID is not)."

**Comment by 0x4007** [comment]
I think let's use strictpascalcase then!

**Comment by gentlementlegen** [comment]
@0x4007 Shall this be applied to other elements too, like variables etc? There is the same version for camel case called `StrictCamelCase`.

**Comment by 0x4007** [comment]
Yes let's apply this across all identifiers.

**Comment by jordan-ae** [comment]
@0x4007, @gentlementlegen  I've added StrictPascalCase to the eslint rules. They'll be a couple of errors in certain files due to the new eslint configuration which I haven't changed because of this comment. I'd love to clean up the files though!

> You also should not change any other files.

I also suggest we create a pre-commit hook. By doing so we can ensure no errors go into the repository and enforce code style. The pre-commit hook will run only on staged files to avoid slow and irrelevant processes.

**Comment by 0x4007** [comment]
We have lint-staged for this purpose but perhaps it's not configured correctly.

**Comment by jordan-ae** [comment]
> We have lint-staged for this purpose but perhaps it's not configured correctly. 

Okay I'll take sometime to look into that too

**Comment by jordan-ae** [comment]
@gentlementlegen please can you review? Thanks

**Review by gentlementlegen — APPROVED** [review]
I am fine with the changes per se, but the changes outside of `.eslintrc` don't seem directly related to this PR but to your own IDE settings.

**Comment by jordan-ae** [comment]
@0x4007 I guess this is ready to go

**Comment by jordan-ae** [comment]
Hey @0x4007 I've undone all the changes not related to the issue. This should be ready to go now!

File: .eslintrc [hunk 2]
```diff
@@ -40,16 +40,16 @@
...
+      { "selector": "memberLike", "modifiers": ["private"], "format": ["StrictPascalCase"], "leadingUnderscore": "require" },
+      { "selector": "typeLike", "format": ["StrictPascalCase"] },
+      { "selector": "typeParameter", "format": ["StrictPascalCase"], "prefix": ["T"] },
+      { "selector": "variable", "format": ["StrictPascalCase"], "leadingUnderscore": "allow", "trailingUnderscore": "allow" },
+      { "selector": "variable", "format": ["StrictPascalCase"], "leadingUnderscore": "allow", "trailingUnderscore": "allow" },
       { "selector": "variable", "modifiers": ["destructured"], "format": null },
       { "selector": "variable", "types": ["boolean"], "format": ["PascalCase"], "prefix": ["is", "should", "has", "can", "did", "will", "does"] },
-      { "selector": "variableLike", "format": ["camelCase"] },
-      { "selector": ["function", "variable"], "format": ["camelCase"] }
+      { "selector": "variableLike", "format": ["StrictPascalCase"] },
```
Comment (gentlementlegen) [inline .eslintrc]: We should not change the `camelCase` to `StrictPascalCase` but `strictCamelCase` instead.

File: .eslintrc [hunk 3]
```diff
@@ -40,16 +40,16 @@
     "sonarjs/no-identical-expressions": "error",
     "@typescript-eslint/naming-convention": [
       "error",
-      { "selector": "interface", "format": ["PascalCase"], "custom": { "regex": "^I[A-Z]", "match": false } },
-      { "selector": "memberLike", "modifiers": ["private"], "format": ["camelCase"], "leadingUnderscore": "require" },
-      { "selector": "typeLike", "format": ["PascalCase"] },
-      { "selector": "typeParameter", "format": ["PascalCase"], "prefix": ["T"] },
-      { "selector": "variable", "format": ["camelCase", "UPPER_CASE"], "leadingUnderscore": "allow", "trailingUnderscore": "allow" },
```
Comment (gentlementlegen) [inline .eslintrc:47]: `UPPER_CASE` should not be removed.

**Review by gentlementlegen — CHANGES_REQUESTED** [review]
Do not make changes that would change the behavior of the configuration. We just need `StrictPascalCase` and `strictCamelCase` over the `PascalCase` and `camelCase` respectively.

**Comment by jordan-ae** [comment]
> I think let's use strictpascalcase then!I 

I thought we were going full in on StrictPascalCase 😬

**Review by gentlementlegen — APPROVED** [review]

**Review by 0x4007 — APPROVED** [review]

**Review by 0x4007 — APPROVED** [review]

**Review by 0x4007 — APPROVED** [review]

**Pull request merged**